packages/base-driver/lib/basedriver/core.ts

```diff
@@ -376,4 +388,58 @@ export class DriverCore<const C extends Constraints, Settings extends StringReco
       this.noCommandTimer = null;
     }
   }
+
+  async executeBidiCommand(bidiCmd: string, bidiParams: StringRecord): Promise<any> {
+    const [moduleName, methodName] = bidiCmd.split('.');
+
+    // if we don't get a valid format for bidi command name, reject
+    if (!moduleName || !methodName) {
+      throw new errors.UnknownCommandError(
+        `Did not receive a valid BiDi module and method name ` +
+          `of the form moduleName.methodName. Instead received ` +
+          `'${moduleName}.${methodName}'`,
+      );
+    }
+
+    // if the command module isn't part of our spec, reject
+    if (!BIDI_COMMANDS[moduleName]) {
+      throw new errors.UnknownCommandError();
+    }
+
+    const {command, params} = BIDI_COMMANDS[moduleName][methodName] as BidiMethodDef;
+    // if the command method isn't part of our spec, also reject
+    if (!command) {
+      throw new errors.UnknownCommandError();
+    }
+
+    // If the driver doesn't have this command, it must not be implemented
+    if (!this[command]) {
+      throw new errors.NotYetImplementedError();
+    }
+
+    const args: any[] = [];
+    if (params?.required?.length) {
+      for (const requiredParam of params.required) {
+        if (_.isUndefined(bidiParams[requiredParam])) {
+          throw new errors.InvalidArgumentError(
+            `The ${requiredParam} parameter was required but you omitted it`,
```

Comment:
I believe we can handle this case better like we do for HTTP commands: https://github.com/appium/appium/blob/4dd95c096e4e4685c4f464b8251370ea001a562a/packages/base-driver/lib/protocol/protocol.js#L219

Reply:
I don't see any reason to be as complicated as we are there?

Reply:
this helps users to fix their requests

Reply:
you could also leave a todo there, so I could improve the check later myself

Reply:
👍